kolibri/core/assets/src/core-actions.js

```diff
@@ -279,11 +287,20 @@ function initContentSession(store, Kolibri, channelId, contentId, contentKind) {
     kind: contentKind,
   }, _contentSessionModel(store));
 
+  if (store.state.core.session.kind[0] === UserKinds.SUPERUSER) {
+    // treat deviceOwner as anonymous user.
+    sessionData.user = null;
+  }
+
   /* Save a new session model and set id on state */
   const sessionModel = ContentSessionLogResource.createModel(sessionData);
-  sessionModel.save().then((newSession) => {
+  const sessionModelPromise = sessionModel.save();
+  promises.push(sessionModelPromise);
+  sessionModelPromise.then((newSession) => {
     store.dispatch('SET_LOGGING_SESSION_ID', newSession.pk);
   });
+
+  return promises;
```

Comment:
It seems a little weird to return an array of promises rather than a single promise. Would probably be cleaner to return `Promise.all(promises)` here, and then the caller doesn't need to do that.